### ISSUE TITLE
Compiler does not warn about redeclared quantifier variables

### ISSUE THREAD

**Issue by psybers** [boalang/compiler#19]
``` java
foreach (i: int; a[i])
    foreach (i : int; b[i])
        ...
```

should signal a semantic check error of redefining i, instead we get a codegen error:

```
Bug.java:138: error: variable ___i is already defined in method map(Project,Mapper<Text,BytesWritable,EmitKey,EmitValue>.Context)
                            for (long ___i = 0; ___i < ___a.length; ___i++)
                                      ^
```
